examples/benchmarks/Makefile

```diff
@@ -2,19 +2,44 @@ JAVAC ?=	javac
 
 BENCHMARKS = dummy nbody
 
-.PHONY: all clean
+BENCH_CFLAGS =    -shared -Wall -fPIC -m64
+BENCH_LDFLAGS =
+C_EXTRA_LDFLAGS_dummy =
+C_EXTRA_LDFLAGS_nbody = -lm
 
-all:
+.PHONY: all c_bench java_bench clean clean_c clean_java
+
+all: c_bench
+
+c_bench:
+	$(foreach i, ${BENCHMARKS}, \
+		echo "Building C benchmark ${i}..."; \
+		cd ${i}/c && \
+		${CC} ${CFLAGS} ${CPPFLAGS} ${BENCH_CFLAGS} -o bench.so bench.c \
+		${LDFLAGS} ${BENCH_LDFLAGS} ${C_EXTRA_LDFLAGS_${i}} || exit $?; \
+		cd ../../; \
+	)
+
+java_bench:
 	$(foreach i, ${BENCHMARKS}, \
 		echo "Building java benchmark ${i}..."; \
 		cd ${i}/java && \
 		CLASSPATH=../../../../iterations_runners/ ${JAVAC} *.java; \
 		cd ../../; \
 	)
 
-clean:
+clean: clean_java clean_c
+
+clean_java:
 	$(foreach i, ${BENCHMARKS}, \
 		cd ${i}/java && \
 		rm *.class; \
-		cd ../../ \
+		cd ../../; \
+	)
+
+clean_c:
+	$(foreach i, ${BENCHMARKS}, \
+		cd  ${i}/c && \
+		rm *.so; \
```

Comment:
If you add a `-f` to `rm` the target will not fail if the shared object does not exist.